### PR TITLE
Feature/incident auto tagger

### PR DESCRIPTION
**Pull request by kevgliss**
This is initial work required for a scheduled auto tagger. It searching through incident resources (google docs at the moment) and analyzes the text, associating known tags with an incident.

I've also added a `discoverable` flag to the tag itself, if the tag should never be automatically associated (for example if it's a common English word). 

The included `nlp.py` is a more generalized version of what is available via the route `service.py` these will be refactored in a separate PR.

### CODE REVIEW

File: src/dispatch/incident/scheduled.py [hunk 1]
```diff
@@ -32,6 +43,47 @@
...
+    tags = tag_service.get_all(db_session=db_session).all()
+    log.debug(f"Fetched {len(tags)} tags from database.")
+
+    tag_strings = [t.name.lower() for t in tags if t.discoverable]
+    phrases = build_term_vocab(tag_strings)
+    matcher = build_phrase_matcher("dispatch-tag", phrases)
+
+    p = plugins.get(
+        INCIDENT_PLUGIN_STORAGE_SLUG
+    )  # this may need to be refactored if we support multiple document types
```
Comment (mvilanova) [inline src/dispatch/incident/scheduled.py:59]: Just adding a note that we will need to refactor this once https://github.com/Netflix/dispatch/pull/223 lands.

File: src/dispatch/incident/scheduled.py [hunk 2]
```diff
@@ -32,6 +43,47 @@
...
+        matched_tags = (
+            db_session.query(Tag)
+            .filter(func.upper(Tag.name).in_([func.upper(t) for t in extracted_tags]))
+            .all()
+        )
+
+        incident.tags.extend(matched_tags)
+        db_session.commit()
+
+        print(f"Associating tags with incident. Incident: {incident.name}, Tags: {extracted_tags}")
```
Comment (mvilanova) [inline src/dispatch/incident/scheduled.py:84]: Remove or replace it with `log.debug()`.

File: src/dispatch/nlp.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
+import logging
```
Comment (mvilanova) [inline src/dispatch/nlp.py:1]: Should we create a directory for nlp and rename this module to service?

Comment (kevgliss) [reply src/dispatch/nlp.py:1]: Not yet, I don't think. A dir with one file seems like overkill, i.e I don't see nlp ever needing view or models.

File: src/dispatch/incident/scheduled.py (hunk 1 above)
Comment (kevgliss) [reply src/dispatch/incident/scheduled.py:59]: Yes eventually, but #223  itself won't require a refactor of this code. We will move each plugin type one by one and refactor all the slug references.

**Pull request merged**